### ISSUE TITLE
Support Redirect Payment for 3rd Party

### ISSUE THREAD

**Issue by SheldonYS** [virtualeconomy/v-wallet-gui#339]
Once 3rd Party website trigger the opening of new tab with 

```
// the parameters invoice, token, callback are optional
https://wallet.v.systems/payment?recipient=xxxx&amount=xxxx&invoice=xxxx&token=xxxx&callback=xxxxxx
```

The web wallet will parse the parameters and do the following these steps:

1. Ask user login (the URL will become https://wallet.v.systems/login?redirect=https%3A%2F%2Fwallet.v.systems%2Fpayment%3Frecipient%3Dxxxx%26amount%3Dxxxx%26invoice%3Dxxxx%26token%3Dxxxx%26callback%3Dxxxxxx)

2. In home page, the payment modal will auto pop-up with `amount`, `recipient`, `description` filled. These fields will be read only. User cannot modify.

3. If pay successfully, wallet will parse `callback` parameter (this parameter could be URL Encode or Base58 Encode). And then async call this URL in background with transaction ID appends at end. (considering 2 cases: Case 1, if url is http://abc.com , the wallet will call http://abc.com?tx_id=xxxxxx . Case 2: if url is http://abc.com/abc?z=xxxx,  the wallet will call http://abc.com/abc?z=xxxx&tx_id=xxxxxx )

**Comment by SheldonYS** [comment]
Requirement is changed. Some 3rd party cannot pass all these parameters. For example, exchange cannot provide `amount` when deposit. So all fields (including `amount`, `recipient` and `invoice`) need change from mandatory to optional. If field is missing, the pop-up modal will let user free input this field.

**Comment by SheldonYS** [comment]
fixed in #369
